plugins/repository-s3/build.gradle

```diff
@@ -311,6 +315,13 @@ if (useFixture) {
     plugin file(tasks.bundlePlugin.archiveFile)
     environment 'AWS_CONTAINER_CREDENTIALS_FULL_URI', { "http://${s3Fixture.addressAndPort}/ecs_credentials_endpoint" }, IGNORE_VALUE
   }
+
+  gradle.taskGraph.whenReady {
+    if (it.hasTask(s3ThirdPartyTests)) {
+      throw new IllegalStateException("Tried to run third party tests but not all of the necessary environment variables 'amazon_s3_access_key', " +
```

Comment:
I don't think this is right. This is going to _always_ throw an exception whenever the task is requested and `useFixture = true`. Where are we actually checking to see that the required properties are set?

Reply:
@mark-vieira `useFixture` can only be `false` if all the properties are set (see the beginning of this file where we read the env variables).

Reply:
If it's true we're by definition not running a 3rd party test and should fail.

Reply:
Got it, so the third party tests expect to hit some external service rather than our embedded fixture?

Reply:
Yea exactly, they run against real S3 and we want to be sure that they will actually run against the real thing and not accidentally just against Minio :)